product_contract/views/sale_order.xml

```diff
@@ -95,17 +95,64 @@
                 </group>
             </xpath>
             <xpath
-                expr="//field[@name='order_line']/tree//field[@name='price_total']"
-                position="after"
+                expr="//field[@name='order_line']/tree//field[@name='price_unit']"
+                position="before"
             >
+                <field name="contract_template_id" column_invisible="1" />
+                <field name="is_contract" column_invisible="1" />
+                <field
+                    name="contract_id"
+                    options='{"no_create": True}'
+                    column_invisible="parent.is_contract == False"
```

Comment:
Shorter, but also more correct, as `is_contract` may be other value than `False`, but also a falsy value.

```suggestion
                    column_invisible="not parent.is_contract"
```

Do the same on the rest.